packages/react-components/src/components/RTE/RichTextEditor.tsx

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import * as React from 'react';
+import { ContentEdit, Watermark } from 'roosterjs-editor-plugins';
+import { Editor } from 'roosterjs-editor-core';
+import { EditorOptions, IEditor } from 'roosterjs-editor-types';
+import {
+  Rooster,
+  createRibbonPlugin,
+  createUpdateContentPlugin,
+  Ribbon,
+  getButtons,
+  RibbonButton,
+  KnownRibbonButtonKey,
+  UpdateMode
+} from 'roosterjs-react';
+import { richTextEditorStyle } from '../styles/RichTextEditor.styles';
+
+/**
+ * Props for {@link RichTextEditor}.
+ *
+ * @beta
+ */
+export interface RichTextEditorProps {
+  content?: string;
+  onChange: (newValue?: string) => void;
+  children: React.ReactNode;
+  placeholderText?: string;
+}
+
+/**
+ * A component to wrap RoosterJS Rich Text Editor.
+ *
+ * @beta
+ */
+export const RichTextEditor: React.FC<RichTextEditorProps> = (props) => {
+  const { content, onChange, children, placeholderText } = props;
+  const editor = React.useRef<IEditor | null>(null);
+  const ribbonPlugin = React.useMemo(() => createRibbonPlugin(), []);
+
+  React.useEffect(() => {
+    if (content !== editor.current?.getContent()) {
+      editor.current?.setContent(content || '');
+    }
+  }, [content]);
+
+  const onRenderRibbon = function (): JSX.Element {
+    const buttons = getButtons([
+      KnownRibbonButtonKey.Bold,
+      KnownRibbonButtonKey.Italic,
+      KnownRibbonButtonKey.BulletedList,
+      KnownRibbonButtonKey.TextColor,
+      KnownRibbonButtonKey.FontSize,
+      KnownRibbonButtonKey.InsertLink
+    ]);
+
+    return (
+      <div style={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center' }}>
+        <Ribbon buttons={buttons as RibbonButton<string>[]} plugin={ribbonPlugin} />
+        <div style={{ flex: 1 }} />
```

Comment:
className prop should be used instead

Reply:
Resolve this later in UI styling ticket

Reply:
It's a small change that will make code cleaner and help to avoid situations that can cause performance issues, as inline styles are not cached by the browser and have to be recalculated on every render.
Let's add it here at the begining so we won't miss it later as the styling task that you mentioned should be about how RTE looks like.
Feel free to delete the styles props at all if you don't need it.